### ISSUE TITLE
Crash on landing in Linux

### ISSUE THREAD

**Issue by abenkovskii** [endless-sky/endless-sky#3041]
Loaded a save file from v0.9.8 in the most recent master (https://github.com/endless-sky/endless-sky/commit/470d52520afe8fdbaaf563d4556290e8122b3548). After flying around a bit got the game to reliably crash.
To reproduce:
1. Compile the current master
2. Load this save file [Igor Ivanov.txt](https://github.com/endless-sky/endless-sky/files/1296556/Igor.Ivanov.txt)
3. [D]epart, and [J]ump
4. After making two jumps your ship will land (flight path is in the save)

Expected: land on a planet as usual
Got: game crashes

Recompiled with AddressSanitizer enabled. Here is its output:
```
=================================================================
==11867==ERROR: AddressSanitizer: heap-buffer-overflow on address 0x603000205088 at pc 0x00000112697f bp 0x7fff8d486de0 sp 0x7fff8d486dd0
READ of size 4 at 0x603000205088 thread T0
    #0 0x112697e in MissionAction::CanBeDone(PlayerInfo const&) const (/home/sky/endless-sky/endless-sky+0x112697e)
    #1 0x109f719 in Mission::Do(Mission::Trigger, PlayerInfo&, UI*) (/home/sky/endless-sky/endless-sky+0x109f719)
    #2 0x13f5f7e in PlayerInfo::Land(UI*) (/home/sky/endless-sky/endless-sky+0x13f5f7e)
    #3 0xeec668 in MainPanel::Step() (/home/sky/endless-sky/endless-sky+0xeec668)
    #4 0x1949e24 in UI::StepAll() (/home/sky/endless-sky/endless-sky+0x1949e24)
    #5 0x42a514 in main (/home/sky/endless-sky/endless-sky+0x42a514)
    #6 0x7fd9fcd23509 in __libc_start_main (/lib64/libc.so.6+0x20509)
    #7 0x42d9c9 in _start (/home/sky/endless-sky/endless-sky+0x42d9c9)

0x603000205088 is located 8 bytes to the left of 31-byte region [0x603000205090,0x6030002050af)
allocated by thread T0 here:
    #0 0x7fd9ffb35158 in operator new(unsigned long) (/lib64/libasan.so.4+0xe0158)
    #1 0x7fd9fe44505e in std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_assign(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) (/lib64/libstdc++.so.6+0x12505e)

SUMMARY: AddressSanitizer: heap-buffer-overflow (/home/sky/endless-sky/endless-sky+0x112697e) in MissionAction::CanBeDone(PlayerInfo const&) const
Shadow bytes around the buggy address:
  0x0c06800389c0: 00 07 fa fa 00 00 00 07 fa fa 00 00 00 00 fa fa
  0x0c06800389d0: 00 00 02 fa fa fa 00 00 01 fa fa fa 00 00 01 fa
  0x0c06800389e0: fa fa 00 00 02 fa fa fa fd fd fd fa fa fa 00 00
  0x0c06800389f0: 02 fa fa fa 00 00 01 fa fa fa 00 00 01 fa fa fa
  0x0c0680038a00: 00 00 02 fa fa fa 00 00 01 fa fa fa 00 00 00 07
=>0x0c0680038a10: fa[fa]00 00 00 07 fa fa 00 00 00 00 fa fa 00 00
  0x0c0680038a20: 00 07 fa fa 00 00 02 fa fa fa 00 00 01 fa fa fa
  0x0c0680038a30: 00 00 01 fa fa fa 00 00 02 fa fa fa fd fd fd fa
  0x0c0680038a40: fa fa 00 00 01 fa fa fa 00 00 01 fa fa fa 00 00
  0x0c0680038a50: 02 fa fa fa 00 00 02 fa fa fa 00 00 01 fa fa fa
  0x0c0680038a60: fd fd fd fa fa fa fd fd fd fa fa fa fd fd fd fa
Shadow byte legend (one shadow byte represents 8 application bytes):
  Addressable:           00
  Partially addressable: 01 02 03 04 05 06 07 
  Heap left redzone:       fa
  Freed heap region:       fd
  Stack left redzone:      f1
  Stack mid redzone:       f2
  Stack right redzone:     f3
  Stack after return:      f5
  Stack use after scope:   f8
  Global redzone:          f9
  Global init order:       f6
  Poisoned by user:        f7
  Container overflow:      fc
  Array cookie:            ac
  Intra object redzone:    bb
  ASan internal:           fe
  Left alloca redzone:     ca
  Right alloca redzone:    cb
==11867==ABORTING
```

**Comment by tehhowch** [comment]
Resolved by #3038

**Comment by endless-sky** [comment]
Fixed by merging #3038.

**Comment by abenkovskii** [comment]
I can confirm that in the commit https://github.com/endless-sky/endless-sky/commit/899829760b9dc2fa7563d973ecb13056d3574ef8 I can no longer reproduce the crash.
